### PR TITLE
fix: some issues related to class

### PR DESCRIPTION
**Pull request by Shanks0224**
- fix compilation error caused by class in closure

### CODE REVIEW

**Comment by Shanks0224** [comment]
@yviansu  PTAL

**Comment by Shanks0224** [comment]
https://github.com/wasm-micro-runtime/ts2wasm/issues/1006

**Review by xujuntwt95329 — APPROVED** [review]
LGTM

File: src/type.ts [hunk 1]
```diff
@@ -2574,6 +2579,7 @@ export class TypeResolver {
                 (func) =>
                     (str =
                         str +
+                        'method_' +
```
Comment (xujuntwt95329) [inline src/type.ts:2582]: @Shanks0224 We fixed this issue in another PR https://github.com/intel/Wasmnizer-ts/pull/59/files#diff-28a5fe490ceac2e568ef26d4663b154deaaffe37ac184152ae6c87817798b656R2580

Could you restore the code here?

Comment (Shanks0224) [reply src/type.ts]: I have rolled back this change.

**Pull request merged**